luatest/server_v2.lua

```diff
@@ -0,0 +1,493 @@
+--   the value for TARANTOOL_HTTP_PORT env variable to be passed into a server
+--   process. Not supported in the default server_v2_instance.lua script.
+-- @int[opt] object.net_box_port Port for net box connection to a new server.
+--   Also, the value for TARANTOOL_LISTEN env variable to be passed into a
+--   server process.
+-- @string[opt] object.net_box_uri URI for net box connection to a new server.
+--   Also, the value for TARANTOOL_LISTEN env variable to be passed into a
+--   server process. Takes precedence over `object.net_box_port`.
+-- @tab[opt] object.net_box_credentials Override the default credentials for net
+--   box connection to a new server.
+-- @tab[opt] object.box_cfg Extra options for `box.cfg()`. Also, the value for
+--   TARANTOOL_BOX_CFG env variable to be passed into a server process.
+-- @return table
+local Server = require('luatest.server'):inherit({})
+
+Server.constructor_checks = fun.chain(
+    Server.constructor_checks,
+    {
+        command = '?string',
+        workdir = '?string',
+        datadir = '?string',
+        engine = '?string',
+        box_cfg = '?table',
+    }
+):tomap()
+Server.vardir = fio.abspath(os.getenv('VARDIR') or DEFAULT_VARDIR)
+
+--- Initialize the server object.
+function Server:initialize()
+    self.id = digest.base64_encode(digest.urandom(9), {urlsafe = true})
+    if self.alias == nil then
+        self.alias = DEFAULT_ALIAS
+    end
+    if self.command == nil then
+        self.command = fio.pathjoin(fio.dirname(fio.abspath(arg[0])), DEFAULT_INSTANCE)
+    end
+    if self.workdir == nil then
+        self.workdir = fio.pathjoin(self.vardir, ('%s-%s'):format(self.alias, self.id))
+        fio.rmtree(self.workdir)
+        fio.mktree(self.workdir)
+    end
+    if self.datadir ~= nil then
+        local ok, err = fio.copytree(self.datadir, self.workdir)
+        if not ok then
+            error(string.format('Failed to copy directory: %s', err))
+        end
+        self.datadir = nil
+    end
+    if self.net_box_port == nil and self.net_box_uri == nil then
+        self.net_box_uri = self.build_listen_uri(self.alias)
+        fio.mktree(self.vardir)
+    end
+    getmetatable(getmetatable(self)).initialize(self)
+end
+
+--- Create a table with env variables on the base of the constructor params.
+-- The result will be passed into a server process.
+-- Env variables to be used:
+--   * TARANTOOL_ALIAS
+--   * TARANTOOL_WORKDIR
+--   * TARANTOOL_HTTP_PORT
+--   * TARANTOOL_LISTEN
+--   * TARANTOOL_BOX_CFG
+--
+-- @return table
+function Server:build_env()
+    local res = getmetatable(getmetatable(self)).build_env(self)
+    if self.box_cfg ~= nil then
+        res.TARANTOOL_BOX_CFG = json.encode(self.box_cfg)
+    end
+    return res
+end
+
+--- Build a server listen URI on the base of the given server alias.
+-- For now, only UNIX sockets are supported.
+--
+-- @string alias Server alias.
+-- @return string
```

Comment:
```suggestion
--- Build a server listen URI based on the given server alias.
-- For now, only UNIX sockets are supported.
--
-- @string alias Server alias.
-- @return string
```

Reply:
I don't understand this part. Does this function build a URI or a server that listens on the given URI?